pcre/pcre.php

```diff
@@ -157,7 +158,7 @@
  * matches given <i>subject</i>, 0 if it does not, or <b>FALSE</b>
  * if an error occurred.
  */
-function preg_match(string $pattern, string $subject, &$matches, int $flags = 0, int $offset = 0): int|false {}
+function preg_match(#[Language('PhpRegExp')] string $pattern, string $subject, &$matches, int $flags = 0, int $offset = 0): int|false {}
```

Comment:
This is not required since IDE insert regexp injection automatically.

Reply:
When that (Language Injection) functionality was originally implemented, PHP attributes didn't exist.

Going forward, it might make sense to implement the injection as an attribute now and remove the old/current functionality later on.

The SQL injection happens by the IDE guesses by checking if strings start with a keyword. Additionally, the SQL flavour is based on scope (which users have to set up). I personally consider this approach to be fundamentally wrong now that we have better ways to hint the language.

I don't have a strong opinion on any of this, my thought is that it may benefit the IDE development.